File: gcj2012R3A.hs
#!/usr/bin/env runghc
import Data.List
import Data.List.Split
main = getContents >>= mapM_ putStrLn . zipWith caseno [1..] . solve . lines
caseno i r = "Case #"++show i++": "++r
solve (c:cs) = map (toString . answer) . take (read c) . analyze $ cs ::[String]
analyze = map (map (map read . words)) . chunksOf 3
toString = unwords . map show
answer [_,l,p] = map (negate . snd) . reverse . sort . zip (zipWith (/) p l) $ [0,-1..]

File: gcj2012R3A.py
#!/usr/bin/env python
from sys import stdin
from operator import div

def solve():
    N = int(stdin.readline())
    L = map(float,stdin.readline().split())
    P = map(float,stdin.readline().split())
    t = reversed(sorted(zip(map(div,P,L),range(0,-N,-1))))
    return " ".join(map(lambda (x,y): str(-y),t))

for x in range(int(stdin.readline())):
    print("Case #"+str(x+1)+": "+str(solve()))
